### PR TITLE
Document CA1052/CA1822 as similar rules

### LINKED ISSUE #488: Document CA1822 as equivalent of MA0036, MA0038 and MA0041
If I'm not mistaken, **CA1822** is an equivalent of **MA0036**, **MA0038** and **MA0041**.
When confirmed, I'll update the doc accordingly.

### PR DESCRIPTION
**Pull request by drieseng**
Document CA1052 as being similar to MA0036.
Document CA1822 as being similar to MA0038 and MA0041.

Fixes #488.

### CODE REVIEW

File: docs/comparison-with-other-analyzers.md [hunk 1]
```diff
@@ -40,9 +40,12 @@
...
 | [CA1000](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1000?WT.mc_id=DT-MVP-5003978) | [MA0018](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0018.md) | CA can also be configured to run against non-public API surface                   |
 | [CA1002](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1002?WT.mc_id=DT-MVP-5003978) | [MA0016](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0016.md) | CA only applies to `List<T>`                                                      |
+| [CA1052](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1052)                        | [MA0036](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0036.md) | CA can be configured to only run against specific API surfaces                    |
```
Comment (meziantou) [inline docs/comparison-with-other-analyzers.md]:
```suggestion
| [CA1052](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1052?WT.mc_id=DT-MVP-5003978)                        | [MA0036](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0036.md) | CA can be configured to only run against specific API surfaces                    |
```

File: docs/comparison-with-other-analyzers.md [hunk 2]
```diff
@@ -40,9 +40,12 @@
...
 | [CA1815](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1815?WT.mc_id=DT-MVP-5003978) | [MA0066](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0066.md) | MA reports only when the struct is used with HashSet types                        |
 | [CA1819](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1819?WT.mc_id=DT-MVP-5003978) | [MA0016](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0016.md) | CA only applies to arrays                                                         |
+| [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822)                        | [MA0038](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0038.md) | CA supports more cases and can be configured to only run against specific API surfaces |
```
Comment (meziantou) [inline docs/comparison-with-other-analyzers.md]:
```suggestion
| [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822?WT.mc_id=DT-MVP-5003978)                        | [MA0038](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0038.md) | CA supports more cases and can be configured to only run against specific API surfaces |
```

File: docs/comparison-with-other-analyzers.md [hunk 3]
```diff
@@ -40,9 +40,12 @@
...
+| [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822)                        | [MA0041](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0041.md) | CA supports more cases and can be configured to only run against specific API surfaces |
```
Comment (meziantou) [inline docs/comparison-with-other-analyzers.md]:
```suggestion
| [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822?WT.mc_id=DT-MVP-5003978)                        | [MA0041](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0041.md) | CA supports more cases and can be configured to only run against specific API surfaces |
```

File: docs/comparison-with-other-analyzers.md (hunk 1 above)
Comment (drieseng) [reply docs/comparison-with-other-analyzers.md]: What's the reason for adding this?

Comment (meziantou) [reply docs/comparison-with-other-analyzers.md]: Microsoft MVPs can get data about links to MS docs based on the `WT.mc_id`

**Pull request merged**